File: DepthFirstSearch.cpp
#include <iostream>
#include "MyArray.h"
#include "BinaryTree.h"
using namespace std;

int main(int argc, char** argv) {
   if (argc != 1) {
      int size = stoi(argv[1]);

      MyArray array(size, false);
      array.generate();
      cout << array << endl;

      BinaryTree tree;
      for (int i = 0; i < size; i++)
         tree.insert(array.array[i]);
      cout << tree.preOrderString() << endl;
      cout << tree.inOrderString() << endl;
      cout << tree.postOrderString() << endl;
   }
}

File: DepthFirstSearch.java
class DepthFirstSearch {
   public static void main(String[] args) {
      if (args.length != 0) {
         int size = Integer.parseInt(args[0]);

         MyArray array = new MyArray(size, false);
         array.generate();
         System.out.println(array);

         BinaryTree tree = new BinaryTree();
         for (int i = 0; i < array.array.length; i++)
            tree.insert(array.array[i]);
         System.out.println(tree.preOrderString());
         System.out.println(tree.inOrderString());
         System.out.println(tree.postOrderString());
      }
   }
}
